### PR TITLE
SONARJAVA-3104 Rule S5344: Authentication should not rely on insecure…

### PR DESCRIPTION
**Pull request by saberduck**
… "PasswordEncoder"

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/security/PasswordEncoderCheck.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  );
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return Arrays.asList(Tree.Kind.METHOD, Tree.Kind.NEW_CLASS);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (tree.is(Tree.Kind.NEW_CLASS) && UNSAFE_PASSWORD_ENCODERS.anyMatch(((NewClassTree) tree))) {
```
Comment (andrea-guarino-sonarsource) [inline java-checks/src/main/java/org/sonar/java/checks/security/PasswordEncoderCheck.java]: I think we should check if we have semantic information
```
    if (!hasSemantic()) {
      return;
    }
```

File: java-checks/src/test/java/org/sonar/java/checks/security/PasswordEncoderCheckTest.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import org.junit.Test;
+import org.sonar.java.checks.security.PasswordEncoderCheck;
+import org.sonar.java.checks.verifier.JavaCheckVerifier;
+
+public class PasswordEncoderCheckTest {
+
+  @Test
+  public void test() {
+    JavaCheckVerifier.verify("src/test/files/checks/security/PasswordEncoder.java", new PasswordEncoderCheck());
```
Comment (andrea-guarino-sonarsource) [inline java-checks/src/test/java/org/sonar/java/checks/security/PasswordEncoderCheckTest.java]: please, add a call to `JavaCheckVerifier.verifyNoIssueWithoutSemantic`

File: java-checks/src/main/java/org/sonar/java/checks/security/PasswordEncoderCheck.java (hunk 1 above)
Comment (saberduck) [reply java-checks/src/main/java/org/sonar/java/checks/security/PasswordEncoderCheck.java]: fixed

File: java-checks/src/test/java/org/sonar/java/checks/security/PasswordEncoderCheckTest.java (hunk 2 above)
Comment (saberduck) [reply java-checks/src/test/java/org/sonar/java/checks/security/PasswordEncoderCheckTest.java]: fixed

**Pull request merged**